modules/opengl/glsl/raycasting.frag

```diff
@@ -135,17 +140,22 @@ vec4 rayTraversal(vec3 entryPoint, vec3 exitPoint, vec2 texCoords, float backgro
 #if defined(INCLUDE_DVR)
         color = APPLY_CHANNEL_CLASSIFICATION(transferFunction, voxel, channel);
         if (color.a > 0) {
-            vec3 gradient =
-                COMPUTE_GRADIENT_FOR_CHANNEL(voxel, volume, volumeParameters, samplePos, channel);
-            gradient = normalize(gradient);
-
+            
+            vec3 gradient;
+            if (first && useNormals) {
```

Comment:
Future: By refactoring what happens inside the loop it would be easier to move this first time-only code outside of the loop

Reply:
yes